phangsPipeline/casaCubeRoutines.py

```diff
@@ -120,19 +182,94 @@ def copy_mask(infile, outfile, huge_cube_workaround=True):
     Copy a mask from infile to outfile. Includes a switch for large cubes, where getchunk/putchunk can segfault
     """
 
-    if huge_cube_workaround:
-        os.system('rm -rf ' + outfile + '/mask0')
-        os.system('cp -r ' + infile + '/mask0' + ' ' + outfile + '/mask0')
+    #if huge_cube_workaround:
+    #    os.system('rm -rf ' + outfile + '/mask0')
+    #    os.system('cp -r ' + infile + '/mask0' + ' ' + outfile + '/mask0')
+    #else:
+    #    myia = au.createCasaTool(casaStuff.iatool)
+    #    myia.open(infile)
+    #    mask = myia.getchunk(getmask=True)
+    #    myia.close()
+    #
+    #    myia = au.createCasaTool(casaStuff.iatool)
+    #    myia.open(outfile)
+    #    mask = myia.putregion(pixelmask=mask)
+    #    myia.close()
+    
+    mask = get_mask(infile)
+    
+    # use putregion to update pixel mask
+    
+    myia = au.createCasaTool(casaStuff.iatool)
+    
+    myia.open(outfile)
+    out_shape = myia.shape() # [X, Y, CHANNEL, [STOKES]]
+    myia.close()
+    
+    assert np.all(mask.shape == out_shape) # input and output shape must be the same
+    
+    myia.open(outfile)
+    
+    no_memory_issue = True
+    out_mask = None
+    if np.prod(out_shape) >= 2880*2880*393: # known memory issue
```

Comment:
Is this threshold for a specific hardware configuration?  It _looks_ like a magic number for a given piece of hardware.

Reply:
This `2880*2880*393` is the known case where the memory issue happens. I'm adding this check to make the code run faster for such known issues. Otherwise it will waste several to tens of minutes to try to run `getchunk` then figure out there is a memory issue. It's not so friendly for code readers indeed, but I think helps to make the code efficient. Not sure if my idea is good or bad though.